plover/gui/main.py

```diff
@@ -273,18 +276,31 @@ def _update_status(self, state):
                 self.connection_ctrl.SetBitmap(self.disconnected_bitmap)
             self.machine_status_sizer.Layout()
         if self.steno_engine.machine:
-            self.status_button.Enable()
-            if self.steno_engine.is_running:
-                self.status_button.SetBitmapLabel(self.on_bitmap)
-                self.SetTitle("%s: %s" % (self.TITLE, self.RUNNING_MESSAGE))
+            if self.steno_engine.machine.state is not STATE_RUNNING:
+                self._show_status(self.STATUS_ERROR)
+            elif self.steno_engine.is_running:
+                self._show_status(self.STATUS_RUNNING)
             else:
-                self.status_button.SetBitmapLabel(self.off_bitmap)
-                self.SetTitle("%s: %s" % (self.TITLE, self.STOPPED_MESSAGE))
+                self._show_status(self.STATUS_STOPPED)
         else:
             self.status_button.Disable()
             self.status_button.SetBitmapLabel(self.off_bitmap)
             self.SetTitle("%s: %s" % (self.TITLE, self.ERROR_MESSAGE))
 
+    def _show_status(self, status):
+        if status is self.STATUS_ERROR:
+            self.status_button.Disable()
+            self.status_button.SetBitmapLabel(self.off_bitmap)
+            self.SetTitle("%s: %s" % (self.TITLE, self.STOPPED_MESSAGE))
+        elif status is self.STATUS_STOPPED:
+            self.status_button.Enable()
+            self.status_button.SetBitmapLabel(self.off_bitmap)
+            self.SetTitle("%s: %s" % (self.TITLE, self.STOPPED_MESSAGE))
+        elif status is self.STATUS_RUNNING:
+            self.status_button.Enable()
+            self.status_button.SetBitmapLabel(self.on_bitmap)
+            self.SetTitle("%s: %s" % (self.TITLE, self.RUNNING_MESSAGE))
+
```

Comment:
This is confusing: why show the stopped message when the machine is not successfully connected? How about this instead:

```diff
diff --git i/plover/gui/main.py w/plover/gui/main.py
index 5013620..bef7878 100644
--- i/plover/gui/main.py
+++ w/plover/gui/main.py
@@ -277,29 +277,33 @@ class MainFrame(wx.Frame):
             self.machine_status_sizer.Layout()
         if self.steno_engine.machine:
             if self.steno_engine.machine.state is not STATE_RUNNING:
-                self._show_status(self.STATUS_ERROR)
+                status = self.STATUS_ERROR
             elif self.steno_engine.is_running:
-                self._show_status(self.STATUS_RUNNING)
+                status = self.STATUS_RUNNING
             else:
-                self._show_status(self.STATUS_STOPPED)
+                status = self.STATUS_STOPPED
         else:
-            self.status_button.Disable()
-            self.status_button.SetBitmapLabel(self.off_bitmap)
-            self.SetTitle("%s: %s" % (self.TITLE, self.ERROR_MESSAGE))
+            status = self.STATUS_ERROR
+        self._show_status(status)

     def _show_status(self, status):
         if status is self.STATUS_ERROR:
-            self.status_button.Disable()
-            self.status_button.SetBitmapLabel(self.off_bitmap)
-            self.SetTitle("%s: %s" % (self.TITLE, self.STOPPED_MESSAGE))
+            enabled = False
+            bitmap = self.off_bitmap
+            message = self.ERROR_MESSAGE
         elif status is self.STATUS_STOPPED:
-            self.status_button.Enable()
-            self.status_button.SetBitmapLabel(self.off_bitmap)
-            self.SetTitle("%s: %s" % (self.TITLE, self.STOPPED_MESSAGE))
+            enabled = True
+            bitmap = self.off_bitmap
+            message = self.STOPPED_MESSAGE
         elif status is self.STATUS_RUNNING:
-            self.status_button.Enable()
-            self.status_button.SetBitmapLabel(self.on_bitmap)
-            self.SetTitle("%s: %s" % (self.TITLE, self.RUNNING_MESSAGE))
+            enabled = True
+            bitmap = self.on_bitmap
+            message = self.RUNNING_MESSAGE
+        else:
+            raise ValueError('invalid status: %s' % str(status))
+        self.status_button.SetBitmapLabel(bitmap)
+        self.SetTitle("%s: %s" % (self.TITLE, message))
+        (self.status_button.Enable if enabled else self.status_button.Disable)()

     def _quit(self, event=None):
         if self.steno_engine:
```

Also, using "stopped" does not seem right, since Plover is still processing strokes in the background. We really need to find another term.